### PR TITLE
Fix EZP-27330: Not possible to see the full name of content item in UDW

### PR DESCRIPTION
**Pull request by adamwojs**
> JIRA: https://jira.ez.no/browse/EZP-27330

# Description

This Patch expands the last selection level in UDW, which allows a user to see the full name of the content item in case of long names. Preview:
 
| Before | https://youtu.be/pEpJFmuLIrA |
|--------|------------------------------|
| **After**  | https://youtu.be/K-J0IyM-hqo |

### CODE REVIEW

**Comment by andrerom** [comment]
_@adamwojs 1.8 is not supported anymore (no normal patches as of 1.9, as is the case [with FT releases](https://doc.ez.no/roadmap)), and 1.9 has had it's last scheduled patch release, so either target 1.7 or master._

**Comment by andrerom** [comment]
As for this solution, I'm unsure what PM had in mind here.

PM review / input ping @rolandbenedetti @SylvainGuittard @inakijv

**Comment by SylvainGuittard** [comment]
What PM proposed in the jira issue was only for the preview box: https://jira.ez.no/browse/EZP-27330?focusedCommentId=210411&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-210411.

Expanding the column is not a bad suggestion. I would prefer to have a resizable column. We did not include this in the jira issue in order to provide a fix to the customer asap.

**Comment by SylvainGuittard** [comment]
> so either target 1.7 (LTS) or master (currently 1.10).
Definitely master

**Comment by adamwojs** [comment]
> I would prefer to have a resizable column. We did not include this in the jira issue in order to provide a fix to the customer asap.

@SylvainGuittard Yes, resizable columns will be the best solution. I've experimented a little bit with CSS `resize` property that we could use to provide this functionality (here is demo: https://jsfiddle.net/ojdw90Lh/1/), but unfortunately IE and Edge not support it yet.

**Comment by andrerom** [comment]
> @SylvainGuittard Yes, resizable columns will be the best solution. I've experimented a little bit with CSS resize property that we could use to provide this functionality (here is demo: https://jsfiddle.net/ojdw90Lh/1/), but unfortunately IE and Edge not support it yet.

That might be an option, Edge will eventually support it so if that can be used somehow in a way PM approves it would be the easiest and most future proof.

**Comment by rolandbenedetti** [comment]
> That might be an option, Edge will eventually support it so if that can be used somehow in a way PM approves, it would be the easiest and most future proof choice.
I also support that option (which was in initial Jince design actually, we just simplified), it's also much better UX for sure than expanding by default the last level, which to me is looking a bit weird.
For the one having a browser not supporting this, well they can get the full title looking at the preview box.

**Comment by inakijv** [comment]
Besides resizable columns, we would like to have the resizer/handle more visible for the user
https://codepen.io/eZ_UI/pen/wdYqVr

**Comment by SylvainGuittard** [comment]
Thanks @inakijv

**Comment by adamwojs** [comment]
I agree that resizer/handler should be more visible for the users, but I'm afraid that now, this solution is compatible with only Webkit-based browsers. This generates two potential problems:

1. In **Firefox** the default resizer/handler is still visible and its overlapping our resizer/handler (what does not look good in my opinion):

zrzut ekranu 2017-05-29 o 11 57 03

2. In **Edge**, column resize not working at all, but the resize handler is visible, what can be confusing for users. 

In conclusion: I think that for now, the cleanest solution is to stay with the default browser resizer/handler, as long as we don't plan to engage javascript in handling column resize:

resizable-uwd-columns

But of course, the choice belongs to PM. Ping @rolandbenedetti,  @inakijv, @SylvainGuittard

**Comment by SylvainGuittard** [comment]
Thanks @adamwojs for testing this solution.
I agree with you this approach is not great for the user experience. 
How much work is it to do it in javascript and make it compatible with all supported browsers?

**Comment by adamwojs** [comment]
@SylvainGuittard I estimate time cost for 1-2 days with an assumption that solution will be UDW specific. I'm not very familiar with YUI which will be the most time-consuming factor in this case, but according to this snippet:  https://jsfiddle.net/RainStudios/mw786v1w/ it looks quite simple to implement.

**Comment by SylvainGuittard** [comment]
@adamwojs Thanks for the feedback and the snippet. 
I don't know if it makes sense to use YUI (even if UDW is built with YUI). If we implement a solution for UDW in javascript (not YUI), we might reuse it for the v2.
What do you think @dpobel ?

**Comment by andrerom** [comment]
> How much work is it to do it in javascript and make it compatible with all supported browsers?
I think this is overdoing it.

@adamwojs Can you open a seperate smaller PR that just fixes the preview box so it shows the full name, and add full name on hover of the link. And then we can let this resizing logic discussion mature a bit and not block the customer requested usability fixes here.

**Comment by adamwojs** [comment]
@andrerom Okay, I'll do it.

**Comment by andrerom** [comment]
@SylvainGuittard This needs a story now, the issue reported by customer was solved in #873

@adamwojs this needs rebase and updated issue number once that is created.

**Comment by SylvainGuittard** [comment]
@andrerom Done!
https://jira.ez.no/browse/EZP-27461

**Comment by bdunogier** [comment]
Given that it is not prioritized, I suggest that we close this PR. It is linked from the issue Sylvain has created, and can be reopened if needed. No disagreements ?

**Comment by andrerom** [comment]
Nope
